pkg/distributor/distributor.go

```diff
@@ -413,6 +417,7 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	const maxExpectedReplicationSet = 5 // typical replication factor 3 plus one for inactive plus one for luck
 	var descs [maxExpectedReplicationSet]ring.InstanceDesc
 
+	streamTrackers := make([]streamTracker, len(streams))
```

Comment:
This slice isn't really necessary, I could just have something like this on line 439:
```go
st = &streamTracker{
    // ...
}
```

And then just pass `st` instead of `&streamTrackers[i]` on line 445. However, the memory usage won't be significantly lower, considering we still have to allocate the `streamTracker{}` objects, but the GC trashing will probably be quite a bit higher, considering how many of these heap objects will be created separately.